File: app/src/main/java/com/commit451/gitlab/model/api/RepositoryFile.java
package com.commit451.gitlab.model.api;

import com.squareup.moshi.Json;

import org.parceler.Parcel;

@Parcel
public class RepositoryFile {
    @Json(name = "file_name")
    String fileName;
    @Json(name = "file_path")
    String filePath;
    @Json(name = "size")
    long size;
    @Json(name = "encoding")
    String encoding;
    @Json(name = "content")
    String content;
    @Json(name = "ref")
    String ref;
    @Json(name = "blob_id")
    String blobId;
    @Json(name = "commit_id")
    String commitId;
    @Json(name = "last_commit_id")
    String lastCommitId;

    public RepositoryFile() {}

    public String getFileName() {
        return fileName;
    }

    public String getFilePath() {
        return filePath;
    }

    public long getSize() {
        return size;
    }

    public String getEncoding() {
        return encoding;
    }

    public String getContent() {
        return content;
    }

    public String getRef() {
        return ref;
    }

    public String getBlobId() {
        return blobId;
    }

    public String getCommitId() {
        return commitId;
    }

    public String getLastCommitId() {
        return lastCommitId;
    }
}

File: app/src/main/java/com/commit451/gitlab/model/api/RepositoryFile.kt
package com.commit451.gitlab.model.api

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@Parcelize
data class RepositoryFile(
    @Json(name = "file_name")
    var fileName: String? = null,
    @Json(name = "file_path")
    var filePath: String? = null,
    @Json(name = "size")
    var size: Long = 0,
    @Json(name = "encoding")
    var encoding: String? = null,
    @Json(name = "content")
    var content: String,
    @Json(name = "ref")
    var ref: String? = null,
    @Json(name = "blob_id")
    var blobId: String? = null,
    @Json(name = "commit_id")
    var commitId: String? = null,
    @Json(name = "last_commit_id")
    var lastCommitId: String? = null
) : Parcelable
